### PR TITLE
[Workspace] Fix: Show a error toast when workspace read only user delete saved objects

### PR DESCRIPTION
**Pull request by yubonluo**
### Description
A circle keeps rendering on the saved objects management page when workspace read only user delete a saved objects. The expected scenario is to show a error toast on this page, and users can continue with other operations.

## Screenshot
Before fix:

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/71875999/989e9d73-09f8-45e0-a6b2-53fe2f6086c9

After fix:

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/71875999/41d61005-37b2-4e87-8bd9-cfeb4936d4d3

## Testing the changes

 - Use the branch to bootstrap and enable workspace feature flag, savedObjects permission feature flag and security plugin
 - Create a workspace named 'source-workspace' and create some saved objects in the current workspace
 - Create a internal user named 'workspace-read-only-user' in security and  assign a read user to workspace-read-only-user in source workspace - setting - user&permission
image

 - Log in as workspace-read-only-user, select source-workspace, and enter the saved objects management page
 - Delete a saved object
## Changelog

- fix: Show error toast when fail to delete saved objects

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx [hunk 1]
```diff
@@ -607,13 +607,21 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
...
+      await Promise.all(deletes);
+      // Unset this
+      this.setState({
+        selectedSavedObjects: [],
+      });
+    } catch (error) {
+      notifications.toasts.addDanger({
+        title: i18n.translate(
+          'savedObjectsManagement.objectsTable.unableDeleteSavedObjectsNotificationMessage',
+          { defaultMessage: 'Unable delete saved objects' }
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:620]:
```suggestion
          { defaultMessage: 'Unable to delete saved objects' }
```

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx [hunk 2]
```diff
@@ -607,13 +607,21 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
...
-      selectedSavedObjects: [],
-    });
-
+    try {
+      await Promise.all(deletes);
+      // Unset this
+      this.setState({
+        selectedSavedObjects: [],
+      });
+    } catch (error) {
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:616]: Shall we just return in the catch? If delete is not successful, there is no need to fetch objects and counts maybe.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx [hunk 3]
```diff
@@ -607,13 +607,21 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
...
+      notifications.toasts.addDanger({
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:617]: Seems the new added code is not covered by unit test.

File: changelogs/fragments/6756.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+fix:
+- Show a error toast when workspace read only user delete saved objects ([#6756](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6756))
```
Comment (SuZhou-Joe) [inline changelogs/fragments/6756.yml:2]:
```suggestion
- Show error toast when fail to delete saved objects ([#6756](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6756))
```
The error toast should be a generic solution when fail to delete saved objects, no need to restrict to workspace permission failure.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx (hunk 2 above)
Comment (yubonluo) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:616]: Sure, I will move the related code to the `try {}`.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx (hunk 3 above)
Comment (yubonluo) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:617]: I will need some unit tests.

**Review by ruanyl — APPROVED** [review]

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:619]: The solution is not good, in such case, the delete modal will be closed no matter if there is error or not.

Comment (yubonluo) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:619]: Make sense. If the user fails to delete the saved objects, the delete modal will continue to display. If the user success successfully deletes the saved objects, the delete modal will not be displayed.

Comment (SuZhou-Joe) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:623]: Would you mind adding a unit test to cover this line? If delete failed, the modal should still be there will notification.toast.addDanger being called and loading is false.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.test.tsx [hunk 5]
```diff
@@ -613,6 +613,42 @@ describe('SavedObjectsTable', () => {
...
+      await new Promise((resolve) => process.nextTick(resolve));
+      // Ensure the state changes are reflected
+      component.update();
+
+      // Set some as selected
+      component.instance().onSelectionChanged(mockSelectedSavedObjects);
+
+      await component.instance().delete();
+
+      expect(notifications.toasts.addDanger).toHaveBeenCalled();
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.test.tsx:650]: Please adding assertions on:
1. The loading state has been changed to false
2. The delete Modal is still display.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx (hunk 2 above)
Comment (yubonluo) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:623]: Sure, I will add some unit tests.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.test.tsx (hunk 5 above)
Comment (yubonluo) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.test.tsx:654]: Sure, I will add these assertions.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**